git/git.go

```diff
@@ -40,6 +40,22 @@ type Ref struct {
 	Sha  string
 }
 
+func (r *Ref) NameOnRemote(remotename string) string {
+	if r.Type != RefTypeLocalBranch {
+		return r.Name
+	}
+
+	if len(remotename) == 0 {
+		return r.Name
+	}
+
+	if RemoteForBranch(r.Name) != remotename {
+		return r.Name
+	}
+
+	return RemoteBranchForLocalBranch(r.Name)
+}
+
```

Comment:
This returns the remote tracking branch name for a local ref, returning the ref name as a fallback. Now, we may want to support the `local:remote` syntax, as seen in commands like `git push heroku heroku:master`. I think that'd be parsed in the command itself and passed to the initialization of the `*Ref`, which would be preferred over `r.Name`.

Thoughts?